spec/integration/can_i_merge_spec.rb

```diff
@@ -35,6 +27,27 @@ module CLI
         let(:invoke_can_i_merge) { subject.can_i_merge }
 
         it "sends a matrix query" do
+          stub_request(:get, "http://pact-broker/matrix?latest=true&latestby=cvp&mainBranch=true&q%5B%5D%5Bpacticipant%5D=Foo&q%5B%5D%5Bversion%5D=1").
+            with(
+              headers: {
+              'Accept'=>'application/hal+json',
+              }).
+            to_return(status: 200, body: File.read("spec/support/matrix.json"), headers: { "Content-Type" => "application/hal+json" })     
+
+          stub_const("ARGV", %w[--pacticipant Foo --version 1])
+          expect($stdout).to receive(:puts).with(/Computer says yes/)
+          invoke_can_i_merge
+        end
+
+
+        it "ignores pacticipant if --ignore flag is provided" do
+          stub_request(:get, "http://pact-broker/matrix?latest=true&latestby=cvp&mainBranch=true&q%5B%5D%5Bpacticipant%5D=Bar&q%5B%5D%5Bversion%5D=3.4.5&ignore%5B%5D%5Bpacticipant%5D=Foo").
+            with(
+              headers: {
+              'Accept'=>'application/hal+json',
+              }).
+            to_return(status: 200, body: File.read("spec/support/matrix.json"), headers: { "Content-Type" => "application/hal+json" })     
```

Comment:
Is this actually testing the ignore logic? I think this is a successful case anyway? Would a more helpful test here to be a matrix result that is failed, but when the collaborating participant is ignored it passes?

Reply:
It's verifying that the request contains `ignore` field.
I tried adding a negative test too, with a failing matrix result, but I'm not sure how to implement it - when I call `invoke_can_i_merge`, that command internally calls `exit` which just kills the test. I'm not sure how to bypass that.